lib/fluent/log.rb

```diff
@@ -53,9 +53,15 @@ def self.str_to_level(log_level_str)
       end
     end
 
-    def initialize(out=STDERR, level=LEVEL_TRACE, opts={})
-      @out = out
-      @level = level
+    def initialize(logger, opts={})
+      # overwrites logger.level= so that config reloading resets level of Fluentd::Log
+      set_logger_orig = logger.class.public_instance_method(:level=).bind(logger)
```

Comment:
Is it not logger, but setter of log level of `logger`, right? If so, use names to mean it.